src/v/serde/serde.h

```diff
@@ -143,7 +144,7 @@ concept is_chrono_duration
 
 template<typename T>
 inline constexpr auto const is_serde_compatible_v
-  = is_envelope<T>
+  = is_not_system_clock_time_point<T> && (is_envelope<T>
```

Comment:
looks like this is not used and this patch can be dropped? or, i mean, we don't support time_point yet so what are we protecting against?

Reply:
> looks like this is not used and this patch can be dropped? or, i mean, we don't support time_point yet so what are we protecting against?

The idea is to prevent somebody from implementing time_point for a clock that isn't a system_clock. For example, `ss::lowres_clock`, because separate instances don't share an epoch.

Reply:
The check is for the future developers ...(this can be refactored to what Ben suggested in his previous comment ...with a static assert).